easybuild/toolchains/gcccuda.py

```diff
@@ -26,15 +26,17 @@
 EasyBuild support for a GCC+CUDA compiler toolchain.
 
 :author: Kenneth Hoste (Ghent University)
+:author: Bart Oldeman (McGill University, Calcul Quebec, Compute Canada)
 """
 
 from easybuild.toolchains.compiler.cuda import Cuda
 from easybuild.toolchains.gcc import GccToolchain
+from easybuild.toolchains.gcccorecuda import GCCcoreCUDA
 
 
 class GccCUDA(GccToolchain, Cuda):
     """Compiler toolchain with GCC and CUDA."""
     NAME = 'gcccuda'
 
     COMPILER_MODULE_NAME = ['GCC', 'CUDA']
-    SUBTOOLCHAIN = GccToolchain.NAME
+    SUBTOOLCHAIN = [GccToolchain.NAME, GCCcoreCUDA.NAME]
```

Comment:
Do we need to be worried about all the older toolchains, i.e. will it work if gcccorecuda is missing for 2019b?

Reply:
Only if recipes try to use it. Unless someone adds a recipe like https://github.com/easybuilders/easybuild-easyconfigs/blob/cf729049d57ef84428077841321f23c3d5cc4131/easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.0.3-gcccuda-2020a.eb

for 2019b, it won't matter that the subtoolchain is not defined for that version.

Reply:
or like this one https://github.com/easybuilders/easybuild-easyconfigs/blob/cf729049d57ef84428077841321f23c3d5cc4131/easybuild/easyconfigs/u/UCX/UCX-1.8.0-gcccorecuda-2020a.eb

Reply:
gcccorecuda.py has  `OPTIONAL = True`, so it will be ignored for 2019b.

The same mechanism is used for e.g. golf, gmkl and iimkl already, and even for gcccore, as older toolchains (talking about 2015a and earlier) didn't have that yet.